modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/detail/Countries.js

```diff
@@ -0,0 +1,164 @@
+import ClayIcon from '@clayui/icon';
+import ClayList from '@clayui/list';
+import PropTypes from 'prop-types';
+import React, {useContext, useMemo, useState} from 'react';
+
+import {StoreStateContext} from '../../context/StoreContext';
+import {numberFormat} from '../../utils/numberFormat';
+
+const KEYWORD_VALUE_TYPE = [
+	{label: Liferay.Language.get('views'), name: 'views'},
+	{label: Liferay.Language.get('views-percentage'), name: 'views-percentage'},
+];
+
+export default function Countries({currentPage}) {
+	const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+	const [keywordValueType, setKeywordValueType] = useState(
+		KEYWORD_VALUE_TYPE[0]
+	);
+
+	const {languageTag, publishedToday} = useContext(StoreStateContext);
+
+	const countries = useMemo(() => {
+		const dataKeys = new Set();
+
+		return currentPage.data.countrySearchKeywords.reduce(
+			(acc, {countryCode, countryName, views, viewsP}) => {
+				if (dataKeys.has(countryCode)) {
+					return acc;
+				}
+
+				dataKeys.add(countryCode);
+
+				return acc.concat({countryCode, countryName, views, viewsP});
+			},
+			[]
+		);
+	}, [currentPage.data.countrySearchKeywords]);
+
+	const handleKeywordValueType = (valueTypeName) => {
+		const newKeywordValueType = KEYWORD_VALUE_TYPE.find(
+			(keywordValueType) => {
+				return keywordValueType.name === valueTypeName;
+			}
+		);
+		setKeywordValueType(newKeywordValueType);
+	};
+
+	return (
+		<>
+			<h5 className="mt-3 sheet-subtitle">
+				{Liferay.Language.get('views-by-location')}
+			</h5>
+
+			<ClayList className="list-group-keywords-list">
+				<ClayList.Item flex>
+					<ClayList.ItemField expand>
+						<ClayList.ItemTitle className="text-truncate-inline">
+							<span className="text-truncate">
+								{Liferay.Language.get('Country')}
+							</span>
+						</ClayList.ItemTitle>
+					</ClayList.ItemField>
+
+					<ClayList.ItemField>
+						<ClayDropDown
+							active={isDropdownOpen}
+							onActiveChange={setIsDropdownOpen}
+							trigger={
+								<ClayButton
+									className="px-0 text-body"
+									displayType="link"
+									small
+								>
+									<span className="font-weight-semi-bold">
+										<span className="pr-2">
+											{keywordValueType.label}
+										</span>
+
+										<ClayIcon symbol="caret-bottom" />
+									</span>
+								</ClayButton>
+							}
+						>
+							<ClayDropDown.ItemList>
+								{KEYWORD_VALUE_TYPE.map((valueType, index) => (
+									<ClayDropDown.Item
+										active={
+											valueType.name ===
+											keywordValueType.name
+										}
+										key={index}
+										onClick={() => {
+											handleKeywordValueType(
+												valueType.name
+											);
+											setIsDropdownOpen(false);
+										}}
+									>
+										{valueType.label}
+									</ClayDropDown.Item>
+								))}
+							</ClayDropDown.ItemList>
+						</ClayDropDown>
+					</ClayList.ItemField>
+				</ClayList.Item>
+
+				{!publishedToday &&
```

Comment:
Why checking if it is published today? Maybe there is something I missed.

Reply:
This is because the CPP doesn't show data from the present day, only for previous days